js-pkg/react/src/main.tsx

```diff
@@ -253,26 +259,40 @@ export function YDocProvider(props: YDocProviderProps) {
     let canceled = false
     let provider: YSweetProvider | null = null
     const doc = new Y.Doc()
-
     ;(async () => {
       provider = await createYjsProvider(doc, docId, authEndpoint, {
         initialClientToken,
         offlineSupport: props.offlineSupport,
       })
 
-      if ((props.showDebuggerLink ?? true) && provider.clientToken) {
-        const url = debuggerUrl(provider.clientToken)
-        console.log(
-          `%cOpen this in Y-Sweet Debugger ⮕ ${url}`,
-          'font-size: 1.5em; display: block; padding: 10px;',
-        )
-        console.log(
-          '%cTo hide the debugger link, pass showDebuggerLink={false} to YDocProvider',
-          'font-style: italic;',
-        )
+      // If the client reconnects, the "connected" event will fire again. We should only show the
+      // debugger link again if the client token has changed.
+      let lastClientToken: { value: ClientToken | null } = { value: null }
+
+      let connectionCallback = (status: YSweetStatus) => {
+        if (props.showDebuggerLink === false) return // note: if showDebuggerLink is undefined, we show the debugger link.
+        if (provider === null) return
+        if (status !== STATUS_CONNECTED) return
+        if (clientTokensEqual(provider?.clientToken ?? null, lastClientToken.value)) return
+        lastClientToken.value = provider.clientToken
+
+        if (provider.clientToken) {
+          const url = debuggerUrl(provider.clientToken)
+          console.log(
+            `%cOpen this in Y-Sweet Debugger ⮕ ${url}`,
+            'font-size: 1.5em; display: block; padding: 10px;',
+          )
+          console.log(
+            '%cTo hide the debugger link, pass showDebuggerLink={false} to YDocProvider',
+            'font-style: italic;',
+          )
+        }
       }
 
+      provider.on('connection-status', connectionCallback)
+
       if (canceled) {
+        provider.off('connection-status', connectionCallback)
         provider.destroy()
```

Comment:
I think the function returned from the `useEffect()` should handle all the cleanup (i.e. destroying the provider). Ideally, destroying the provider would also make sure all event subscriptions are removed. The original purpose of the `canceled` check here, I think, is to drop out of the execution following an `await` in case the promise chain had been canceled in the meantime.

In other words, I think we should (1) make sure `provider.destroy()` removes all event subscriptions, (2) remove the logic from this `if (canceled)` block except the `return`.